crates/ironrdp-web/src/session.rs

```diff
@@ -216,13 +205,28 @@ impl SessionBuilder {
         self.clone()
     }
 
-    /// Optional
-    pub fn use_display_control(&self) -> SessionBuilder {
-        self.0.borrow_mut().use_display_control = true;
+    pub fn extension(&self, value: JsValue) -> SessionBuilder {
+        match serde_wasm_bindgen::from_value::<Extension>(value) {
+            Ok(value) => match value {
+                Extension::KdcProxyUrl(kdc_proxy_url) => self.0.borrow_mut().kdc_proxy_url = Some(kdc_proxy_url),
+                Extension::Pcb(pcb) => self.0.borrow_mut().pcb = Some(pcb),
+                Extension::DisplayControl(use_display_control) => {
+                    self.0.borrow_mut().use_display_control = use_display_control
+                }
+            },
+            Err(err) => error!("Provided JsValue is not a valid extension value: {err:?}"),
```

Comment:
https://github.com/Devolutions/IronRDP/blob/401cedf01014d8400b460911764d9ebc837900fc/STYLE.md#logging

```suggestion
            Err(error) => error!(%error, "Provided JsValue is not a valid extension value"),
```